overlord/snapstate/handlers_components.go

```diff
@@ -424,6 +424,11 @@ func (m *SnapManager) doLinkComponent(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// during a revert, components should already be linked
+	if snapsup.Revert {
```

Comment:
any reason not to skip adding the tasks instead in the first place?

Reply:
Good question, I feel like I originally had a reason for taking this approach, but it doesn't seem relevant now. I've reworked this to not create the tasks.